### ISSUE TITLE
False positive 'unresolved reference'

### ISSUE THREAD

**Issue by CicadaCinema** [intellij-rust/intellij-rust#7997]
## Environment

* **IntelliJ Rust plugin version:** `0.4.157.4174-212`
* **Rust toolchain version:** `stable-x86_64-unknown-linux-gnu (default)` `rustc 1.56.0 (09c42c458 2021-10-18)`
* **IDE name and version:** `CLion 2021.2.3` `Build #CL-212.5457.51`
* **Operating system:** Fedora Linux 34 `Linux 5.14.12-200.fc34.x86_64`

## Problem description

An import is not being recognised by the plugin, despite the code building and running fine. Not sure what the exact cause of the issue is in this case, but I believe problems like this need to be tracked at this point. I saw this referenced in https://github.com/intellij-rust/intellij-rust/issues/3116 3 years ago and I think this plugin has evolved enough since then to allow me to trust the `Unresolved reference` error annotation.

![image](https://user-images.githubusercontent.com/52425971/138574385-9d2bc7ae-1137-41dd-bfbc-c472bdf333e9.png)

## Steps to reproduce

The following minimal 'example' is enough to reproduce the problem:

`cargo.toml`
```rust
[package]
name = "bug_poc"
version = "0.1.0"
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
url = "2.2.2"
matrix-sdk = { git = "https://github.com/matrix-org/matrix-rust-sdk", rev = "9e1024f4b5" }
tracing-subscriber = "0.2.25"
tokio = { version = "1.12.0", features = ["full"] }
```

`main.rs`
```rust
use std::{env, process::exit};

use matrix_sdk::{
    self,
    config::SyncSettings,
    room::Room,
    ruma::events::room::message::{
        MessageEventContent, MessageType, SyncMessageEvent, TextMessageEventContent,
    },
    Client,
};
use url::Url;

async fn on_room_message(event: SyncMessageEvent, room: Room) {
    if let Room::Joined(room) = room {
        if let SyncMessageEvent {
            content:
            MessageEventContent {
                msgtype: MessageType::Text(TextMessageEventContent { body: msg_body, .. }),
                ..
            },
            sender,
            ..
        } = event
        {
            let member = room.get_member(&sender).await.unwrap().unwrap();
            let name = member.display_name().unwrap_or_else(|| member.user_id().as_str());
            println!("{}: {}", name, msg_body);
        }
    }
}

async fn login(
    homeserver_url: String,
    username: &str,
    password: &str,
) -> Result<(), matrix_sdk::Error> {
    let homeserver_url = Url::parse(&homeserver_url).expect("Couldn't parse the homeserver URL");
    let client = Client::new(homeserver_url).unwrap();

    client.register_event_handler(on_room_message).await;

    client.login(username, password, None, Some("rust-sdk")).await?;
    client.sync(SyncSettings::new()).await;

    Ok(())
}

#[tokio::main]
async fn main() -> Result<(), matrix_sdk::Error> {
    tracing_subscriber::fmt::init();

    let (homeserver_url, username, password) =
        match (env::args().nth(1), env::args().nth(2), env::args().nth(3)) {
            (Some(a), Some(b), Some(c)) => (a, b, c),
            _ => {
                eprintln!(
                    "Usage: {} <homeserver_url> <username> <password>",
                    env::args().next().unwrap()
                );
                exit(1)
            }
        };

    login(homeserver_url, &username, &password).await
}
```

**Comment by mili-l** [comment]
see https://github.com/intellij-rust/intellij-rust/pull/9729
